lsp-semantic-tokens.el

```diff
@@ -541,6 +577,144 @@ LOUDLY will be forwarded to OLD-FONTIFY-REGION as-is."
                when (> current-line line-max-inclusive) return nil)))))
       `(jit-lock-bounds ,beg . ,end)))))
 
+(defun lsp-semantic-tokens-toggle-comment-syntax ()
+  "Toggle the value of `lsp-semantic-tokens-set-comment-syntax'"
+  (interactive)
+  (if lsp-semantic-tokens-set-comment-syntax
+      (progn
+        (setq lsp-semantic-tokens-set-comment-syntax nil)
+        (lsp-semantic-tokens--remove-comment-syntax (point-min) (point-max)))
+    (setq lsp-semantic-tokens-set-comment-syntax t))
+  (font-lock-flush))
+
+(defun lsp-semantic-tokens--get-overlapping-comments (beg end)
```

Comment:
with the way this is used, I think it'd make sense to always return non-nil `beg` and `end`, with `beg` or `end` remaining unmodified if they don't happen to overlap with comment tokens. Maybe the function could also be renamed to something like `extend-region-to-include-comment-tokens` or something like that (plus the prefix, of course), in my opinion that would be slightly more intuitive